File: cs/fillArray.cs
using System;

// Algorithm which fills an array (pre-defined length) with integers
class FillArray
{
    static void Main(string[] args)
    {
        int[] intArray;
        var maxNum = 1000000; // by default 1 million integers in array
        var stopwatch = new System.Diagnostics.Stopwatch();

        if (args.Length != 0)
        {
            Console.WriteLine("Parameter (string) provided, parsing to integer: " + args[0]);
            maxNum = Int32.Parse(args[0]);
        }
        else
        {
            Console.WriteLine("No parameter provided, using default {0}", maxNum);
        }

        intArray = new int[maxNum];
        stopwatch.Start();

        for (int i = 0; i < maxNum; i++)
        {
            intArray[i] = i;
        }

        stopwatch.Stop();
        Console.WriteLine("Time elapsed: {0}{1}", stopwatch.ElapsedMilliseconds, "ms");
    }
}

File: java/fillArray.java
/**
 * Algorithm which fills an array (fixed length) with numbers
 */
class fillArray {
  public static void main(String[] args) {
    long startTime;
    int[] intArray;
    int maxNum = 1000000; // by default 1 million integers in array

    if (args.length != 0) {
      System.out.println("Parameter (String) provided, parsing to integer: " + args[0]);
      maxNum = Integer.parseInt( args[0] );
    } else {
      System.out.println("No parameter provided, using default " + maxNum);
    }

    intArray = new int[maxNum];
    startTime = System.nanoTime();

    for (int i = 0; i < maxNum; i++) {
      intArray[i] = i;
    }

    System.out.println(
      "Time " + (float) (System.nanoTime() - startTime) / 1000000000 + "s"
    );
  }
}

/*
Intel i7 2.9 GHz, Windows 10 64bit, Java OpenJDK 15.0.1

3 test runs with parameters: 1000002, 1000003, 1000004

0.0051042s (5.1ms)
0.0049314s (4.9ms)
0.0058498s (5.8ms)
AVERAGE: 5.3ms

*/
